File: ios/ViewManagers/SmileIDSmartSelfieCaptureViewManager.swift
import Foundation
import React
import SmileID
import SwiftUI

@objc(SmileIDSmartSelfieCaptureViewManager)
class SmileIDSmartSelfieCaptureViewManager: SmileIDBaseViewManager {
    override func getView() -> UIView {
        BaseSmileIDView(frame: .zero, contentView: AnyView(
            SmileIDSmartSelfieCaptureView(
                product: product,
                smileIDUIViewDelegate: self
            )),
        product: product)
    }

    @objc func setParams(_ node: NSNumber, commandId _: NSNumber, params: NSDictionary) {
        /*  UI Updates on the Main Thread:async ensures that the UI update is scheduled to run on the next cycle of the run loop, preventing any potential blocking of the UI if the update were to take a noticeable amount of time
         */
        DispatchQueue.main.async {
          var smileSensitivity: String?
          if let smileSensitivityValue = params["smileSensitivity"] as? String {
            smileSensitivity = smileSensitivityValue
          }
          if self.bridge.uiManager.view(forReactTag: node) is BaseSmileIDView {
                self.product.allowAgentMode = params["allowAgentMode"] as? Bool ?? false
                self.product.forceAgentMode = params["forceAgentMode"] as? Bool ?? false
                self.product.userId = params["userId"] as? String
                self.product.jobId = params["jobId"] as? String
                self.product.showConfirmation = params["showConfirmation"] as? Bool ?? true
                self.product.showInstructions = params["showInstructions"] as? Bool ?? true
                self.product.smileSensitivity = smileSensitivity?.toSmileSensitivity() ?? .normal
                self.product.showAttribution = params["showAttribution"] as? Bool ?? true
                self.product.useStrictMode = params["useStrictMode"] as? Bool ?? false
                self.product.onResult = params["onResult"] as? RCTBubblingEventBlock
            }
        }
    }
}

File: android/src/main/java/com/smileidentity/react/viewmanagers/SmileIDSmartSelfieCaptureViewManager.kt
package com.smileidentity.react.viewmanagers

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.smileidentity.react.toSmileSensitivity
import com.smileidentity.react.utils.getBoolOrDefault
import com.smileidentity.react.utils.getStringOrDefault
import com.smileidentity.react.views.SmileIDSmartSelfieCaptureView
import com.smileidentity.react.views.SmileIDSmartSelfieEnrollmentEnhancedView
import com.smileidentity.react.views.SmileIDView

@ReactModule(name = SmileIDSmartSelfieCaptureViewManager.NAME)
class SmileIDSmartSelfieCaptureViewManager(
  private val reactApplicationContext: ReactApplicationContext
) : BaseSmileIDViewManager<SmileIDSmartSelfieCaptureView>(reactApplicationContext) {
  override fun getName(): String = NAME

  override fun createSmileView(): SmileIDSmartSelfieCaptureView {
    return SmileIDSmartSelfieCaptureView(reactApplicationContext)
  }

  override fun applyArgs(view: SmileIDSmartSelfieCaptureView, args: ReadableMap?) {
    args?.let {
      view.userId = it.getStringOrDefault("userId")
      view.jobId = it.getStringOrDefault("jobId")
      view.allowAgentMode = it.getBoolOrDefault("allowAgentMode", false)
      view.forceAgentMode = it.getBoolOrDefault("forceAgentMode", false)
      view.showAttribution = it.getBoolOrDefault("showAttribution", true)
      view.showInstructions = it.getBoolOrDefault("showInstructions", true)
      view.useStrictMode = it.getBoolOrDefault("useStrictMode", false)
      view.showConfirmation =
        it.getBoolOrDefault("showConfirmation", true)
      view.smileSensitivity =
        it.getStringOrDefault("smileSensitivity", null)?.toSmileSensitivity()
    }
  }

  companion object {
    const val NAME = "SmileIDSmartSelfieCaptureView"
  }
}
